src/Routing/RouteCollection.php

```diff
@@ -425,6 +432,47 @@ public function registerMiddleware($name, callable $middleware)
     }
 
     /**
+     * Add middleware to a middleware group
+     *
+     * @param string $name Name of the middleware group
+     * @param array $middlewareNames Names of the middleware
+     * @return $this
+     */
+    public function middlewareGroup($name, $middlewareNames)
+    {
+        if ($this->hasMiddleware($name)) {
+            $message = "Cannot add middleware group '$name'. A middleware by this name has already been registered.";
+            throw new RuntimeException($message);
+        }
+        if ($this->hasMiddlewareGroup($name)) {
```

Comment:
Should it be possible to overwrite a middleware group? Right now once defined middleware groups are immutable.